From: SeongJae Park <sj@kernel.org>
Date: Fri, 30 Jan 2026 17:29:21 -0800
Subject: [PATCH] mm/damon/core: adjust end address on walk_system_ram() for
 inclusiveness

'struct damon_addr_range' is for end-open ranges ([start, end)), while
'struct resource' uses fully-inclusive ranges ([start, end]).  But
walk_system_ram() is assigning resource->end to damon_addr_range->end
without adjustment.  As a result, the monitoring target regions miss the
last one byte.  Shouldn't be a real problem, but a bug is a bug.  Do the
adjustment by adding value one to the end address.

Fixes: 43b0536cb471 ("mm/damon: introduce DAMON-based Reclamation (DAMON_RECLAIM)")
Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index aac49ec5c8ebe..f56e54b650019 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -3421,7 +3421,7 @@ static int walk_system_ram(struct resource *res, void *arg)
 
 	if (a->end - a->start < resource_size(res)) {
 		a->start = res->start;
-		a->end = res->end;
+		a->end = res->end + 1;
 	}
 	return 0;
 }
-- 
2.47.3

